src/main/java/org/cardanofoundation/cfexploreraggregator/uniqueaccount/processor/UniqueAccountProcessor.java

```diff
@@ -66,28 +86,28 @@ public void handleCommitEvent(CommitEvent commitEvent) {
         Set<String> snapshot = new HashSet<>(uniqueAccounts);
         uniqueAccounts.clear();
         snapshot.forEach(address -> {
-            Optional<AccountInEpochEntity> byAddressAndEpoch = addressInEpochRepository.findByAddressAndEpoch(address, commitEvent.getMetadata().getEpochNumber());
-            List<AccountInEpochEntity> accountsInEpoch = new ArrayList<>();
+            Optional<UniqueAddressesInEpochEntity> byAddressAndEpoch = uniqueAddressesInEpochRepository.findByAddressAndEpoch(address, commitEvent.getMetadata().getEpochNumber());
+            List<UniqueAddressesInEpochEntity> accountsInEpoch = new ArrayList<>();
             if(byAddressAndEpoch.isEmpty()) {
-                accountsInEpoch.add(AccountInEpochEntity.builder()
+                accountsInEpoch.add(UniqueAddressesInEpochEntity.builder()
                         .address(address)
                         .epoch(commitEvent.getMetadata().getEpochNumber())
                         .slot(commitEvent.getMetadata().getSlot())
                         .build());
             }
-            addressInEpochRepository.saveAll(accountsInEpoch);
+            uniqueAddressesInEpochRepository.saveAll(accountsInEpoch);
         });
```

Comment:
If I'm reading this properly you're checking one-by-one against db.

Not mandatory, but I might suggest: fetch all addresses by epoch in 1 go, filter and save remaining.

Good usage of slot, if I'm reading this fine you're saving the first occurrence the address was seen? This hsould be rollback resilient

Reply:
Done